components/sensor/atm90e32.rst

```diff
@@ -0,0 +1,193 @@
+
+The atm90e32 IC can measure up to three AC voltages although typically only one
+voltage measurement would be used for the mains electricity phase of a
+household. Three current measurements are read via CT clamps. 
+
+The `CircuitSetup 2-Channel Energy Monitor <https://circuitsetup.us/index.php/product/split-single-phase-real-time-whole-house-energy-meter-v1-2/>`__ can read 2 current channels and one voltage channel.
+
+.. figure:: images/atm90e32-cs-2chan-full.jpg
+    :align: center
+    :width: 50.0%
+
+    CircuitSetup Split Single Phase Real Time Whole House Energy Meter.
+
+The `CircuitSetup 6-Channel Energy Monitor <https://circuitsetup.us/index.php/product/expandable-6-channel-esp32-energy-meter/>`__ can read 6 current channels and 2 voltage channels at a time, this board has two atm90e32 ICs and requires two sensors to be configured in ESPHome.
+
+.. figure:: images/atm90e32-cs-6chan-full.jpg
+    :align: center
+    :width: 50.0%
+
+    CircuitSetup Expandable 6 Channel ESP32 Energy Meter Main Board.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+      - platform: atm90e32
+        cs_pin: 5
+        phase_a:
+          voltage:
+            name: "EMON Line Voltage A"
+          current:
+            name: "EMON CT1 Current"
+          power:
+            name: "EMON Active Power CT1"
+          gain_voltage: 41820
+          gain_ct: 25498
+        phase_b:
+          current:
+            name: "EMON CT2 Current"
+          power:
+            name: "EMON Active Power CT2"
+          gain_voltage: 41820
+          gain_ct: 25498
+        frequency:
+          name: "EMON Line Frequency"
+        line_freq: 50Hz
+        gain_pga: 2X
+        update_interval: 60s
+
+Configuration variables:
+------------------------
+
+- **cs_pin** (**Required**, :ref:`Pin Schema <config-pin_schema>`): The pin CS is connected to.
+- **phase_a** (*Optional*): The configuration options for the 1st phase.
+
+  - **voltage** (*Optional*): Use the voltage value of this phase in V (RMS).
+    All options from :ref:`Sensor <config-sensor>`.
+  - **current** (*Optional*): Use the current value of this phase in amperes. All options from
+    :ref:`Sensor <config-sensor>`.
+  - **power** (*Optional*): Use the power value on this phase in watts. All options from
+    :ref:`Sensor <config-sensor>`.
+  - **gain_voltage** (*Optional*, int): Voltage gain to scale the low voltage AC power pack to household mains feed.
+    Defaults to ``41820``.
+  - **gain_ct** (*Optional*, int): CT clamp calibration for this phase.
+    Defaults to ``25498``.
+
+- **phase_b** (*Optional*): The configuration options for the 2nd phase. Same options as 1st phase.
+- **phase_c** (*Optional*): The configuration options for the 3rd phase. Same options as 1st phase.
+- **line_freq** (*Required*, string): The AC line frequency of the supply voltage. One of ``50Hz``, ``60Hz``
```

Comment:
```suggestion
- **line_freq** (**Required**, string): The AC line frequency of the supply voltage. One of ``50Hz``, ``60Hz``
```

See last PR comment - and move this up to top please because it's a required option.